programs/application/src/lib.rs

```diff
@@ -28,20 +40,30 @@ pub mod application {
         ctx: Context<UpdateStatus>,
         _application_id: String,
         _application_bump: u8,
-        status: bool,
+        status: u8,
     ) -> Result<()> {
-        if status {
-            ctx.accounts.base_account.status = JobStatus::Selected;
-        } else {
-            ctx.accounts.base_account.status = JobStatus::Rejected;
+
+        match status {
```

Comment:
can we implement `AnchorSerialize` and `AnchorDeserialize` for the enum? It would be more idiomatic

Reply:
oh, actually it is. So maybe instead of passing an u8, just pass `JobStatus`

Reply:
Isnt it implemented in the application program when we defined the enum?

Reply:
yes but then let's use it instead of a u8

Reply:
But how do u send it from the client side?

Reply:
you probably send a number from client side. But I don't think you need to go through the match statement if there's already an `AnchorDeserialize` implemented.

Reply:
im sending a number, im getting an error

Reply:
lemme check